atomic_reactor/plugins/exit_verify_media_types.py

```diff
@@ -0,0 +1,148 @@
+        if self.workflow.push_conf.pulp_registries:
+            raise KeyError("pulp registry configure, verify_media_types should not run")
+        image = self.workflow.tag_conf.unique_images[0]
+
+        media_types = set()
+        registries = deepcopy(get_registries(self.workflow, {}))
+        for registry_name in registries:
+            registry = registries[registry_name]
+            initial_media_types = registry.get('expected_media_types', [])
+            if not initial_media_types:
+                continue
+
+            expected_media_types = self.set_manifest_list_expectations(initial_media_types)
+
+            pullspec = image.copy()
+            pullspec.registry = registry_name
+            insecure = registry.get('insecure', False)
+            secret = registry.get('secret', None)
+
+            check_digests = (MEDIA_TYPE_DOCKER_V2_MANIFEST_LIST in expected_media_types or
+                             MEDIA_TYPE_DOCKER_V2_SCHEMA2 in expected_media_types or
+                             MEDIA_TYPE_DOCKER_V2_SCHEMA1 in expected_media_types)
+            if check_digests:
+                digests = get_manifest_digests(pullspec, registry_name, insecure, secret,
+                                               require_digest=False)
+                if digests:
+                    if digests.v2_list:
+                        self.log.info("Manifest list found")
+                        if MEDIA_TYPE_DOCKER_V2_MANIFEST_LIST in expected_media_types:
+                            media_types.add(MEDIA_TYPE_DOCKER_V2_MANIFEST_LIST)
+                    if digests.v2:
+                        self.log.info("V2 schema 2 digest found")
+                        if MEDIA_TYPE_DOCKER_V2_SCHEMA2 in expected_media_types:
+                            media_types.add(MEDIA_TYPE_DOCKER_V2_SCHEMA2)
+                    if digests.v1:
+                        self.log.info("V2 schema 1 digest found")
+                        if MEDIA_TYPE_DOCKER_V2_SCHEMA1 in expected_media_types:
+                            media_types.add(MEDIA_TYPE_DOCKER_V2_SCHEMA1)
+
+            if MEDIA_TYPE_DOCKER_V1 in expected_media_types:
+                if verify_v1_image(pullspec, registry_name, self.log, insecure, secret):
+                    media_types.add(MEDIA_TYPE_DOCKER_V1)
+
+            # sorting the media type here so the failure message is predictable for unit tests
+            for media_type in sorted(expected_media_types):
+                if media_type not in media_types:
+                    raise KeyError("expected media type '{0}' ".format(media_type) +
+                                   "not in available media types {0}".format(media_types))
+        final_media_types = [media_type for media_type in media_types]
+        return sorted(final_media_types)
+
+    def set_manifest_list_expectations(self, expected_media_types):
+        if self.workflow.postbuild_results.get(PLUGIN_GROUP_MANIFESTS_KEY):
```

Comment:
nitpick: if you reverse this check, you don't have to nest most of this method's body:
```python
if not self.workflow.postbuild_results.get(PLUGIN_GROUP_MANIFESTS_KEY):
  return expected_media_types

platforms = get_platforms(self.workflow)
...
```

Reply:
Inherited the function from pulp_pull, but fixed.